spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDParser.java

```diff
@@ -43,4 +45,29 @@
 	 *         described by the DSL.
 	 */
 	List<ModuleDescriptor> parse(String name, String config, ParsingContext type);
+
+	public static class DefinitionOrException {
```

Comment:
When I said "transport changes", I meant something other than http for the API (theoretical consideration, obviously). My rule of thumb being that we should be able _e.g._ to accept commands over let's say raw telnet and the underlying business logic should not change. Sorry for the use of (yet another) overloaded term.

The transient repository is here not just to avoid hitting the real repository, but also to span the whole method call. In the example I gave above, it stores the first successful stream "creation" (although we don't save it for real) so that _e.g._ a `tap:stream:foo` would succeed. This is also why it's not just about moving the loop to the controller (or the javascript client, for that matter).
I couldn't agree more with you about the parser, but this extension merely acknowledges the fact that we now want to support the 
```
name = DSL
name = DSL
name = DSL
```
in addition to the simple
```
DSL
```
construct. 

About the class names, those are definitely non final and the aim of this PR is exactly what we're doing now, _i.e._ discuss the approach.